src/MLJInterface.jl

```diff
@@ -170,11 +170,30 @@ function fit(mlj_model::MODELS, verbosity::Int, X, y, w=Vector{AbstractFloat}())
     w = Float32.(w)
     report = LightGBM.fit!(model, X, y_lgbm; verbosity=verbosity, weights=w)
 
-    model = (model, classes)
+    fitresult = (model, classes, mlj_model)
     cache = nothing
     report = (report,)
 
-    return (model, cache, report)
+    return (fitresult, cache, report)
+
+end
+
+function update(mlj_model::MLJInterface.MODELS, verbosity::Int, fitresult, cache, X, y, w=AbstractFloat[])
+
+    old_lgbm_model, old_classes, old_mlj_model = fitresult
+
+    # we can continue boosting if and only if num_iterations has changed
+    if MLJModelInterface.is_same_except(old_model, mode, :num_iterations)
+        additional_iterations = mlj_model.num_iterations - old_mlj_model.num_iterations
+        # eh this is ugly, possibly prompts a need for some refactoring
```

Comment:
I'd check this is non-negative and otherwise flip to a cold restart (`fit` call).

Reply:
25ee0d4